### DISCUSSION TITLE
Random TCP port appear in URL after server migration

### DISCUSSION THREAD

**Discussion by jcarnat** [pixelfed/pixelfed#4871] [Q&A]
Hi,

I want to move my instance from one server to another. I have reinstalled the prerequisites on the new server and configured Pixelfed as if it was new. Then I restored the content of a backup on that new server and restarted the new server.

There are a few weird things happening like images not rendering. But the weirdest thing is that the page have `href` containning a random port in the URL.
```
# curl -s https://chicklens.foolbazar.eu | grep 'https://'
<link rel="manifest" href="https://chicklens.foolbazar.eu:45243/manifest.json" />
<link rel="icon" type="image/png" href="https://chicklens.foolbazar.eu:45243/img/favicon.png">
<link rel="apple-touch-icon" type="image/png" href="https://chicklens.foolbazar.eu:45243/img/favicon.png">
<link rel="preload" as="image" href="https://chicklens.foolbazar.eu:45243/_landing/bg.jpg" />
```
Each call generates a different port. Those ports are not accessible as only https:443 is exposed.
If that matters, nginx is 1.24 and php 8.1.27.

Any idea where this might come from and how to solve this?

Thanks.

**Comment by x64x2** [comment]
Inspect config files, clear cache, and verify db entries for incorrectly hardcoded ports. Adjust Nginx and Pixelfed settings accordingly.

**Comment by jcarnat** [comment]
Right, there were two issues :
- My reverse-proxy had a weird configuration.
Ensuring `X-Forwarded-Proto` and `X-Forwarded-Port` were properly setup solved the port issue.
- The ZIP backup contains a src/public/storage restored as a (empty) directory.
I had to `rmdir` it and run `php artisan storage:link`.
